(FM-8503) added transport for test device in order to be able to test transport for ace

Lavinia-Dan opened the pull request.

Lavinia-Dan committed:
FM-8503 added transport for test device

DavidS reviewed (changes requested):
- [x] The existing `spinner` type and provider should also work with the new transport, no need to duplicate those
- [x] The `spinner_transport` schema must define the same values used by the existing `spinner` device (see all the `config` accesses in https://github.com/puppetlabs/puppetlabs-test_device/blob/master/lib/puppet/util/network_device/spinner/device.rb)
- [x] do not delete the pre-rendered unit tests from the PDK, but use them as a basis for basic checks (if they're at all salvageable).
- [x] use the `spinner_transport/device.rb` for local testing with `puppet device`, but rename it to `device.rb.example` for the commit, so it does not interfere with our long-term goals of testing a module with no `device.rb`
- [x] copy/paste the results of a manual test-run in the commit message to confirm that you could successfully run it
- [x] optional extra cleanup: extract the common code from `spinner` device and `spinner_transport` transport into a `module` used by both.

Lavinia-Dan committed:
FM-8503 fixed rubocop offenses

Lavinia-Dan committed:
unit tests for transport, clean ups

Lavinia-Dan committed:
cleanups

Lavinia-Dan committed:
fix rubocop offenses

Lavinia-Dan commented:
![Screenshot 2019-09-27 at 15 09 09](https://user-images.githubusercontent.com/5837544/65768171-d6d1c280-e138-11e9-948d-a371f1e76807.png)

Lavinia-Dan committed:
fixes

Lavinia-Dan committed:
fixed tests

Lavinia-Dan committed:
cleanup and fixed rubocop offenses

Lavinia-Dan committed:
renamed spinner_transport/device.rb to device.rb.example so it won't interfere with lng-term goals of testing a module with no device

DavidS committed:
(FM-8503) Fix the task to run with both device and transport

This adds some functionality from https://github.com/puppetlabs/puppetlabs-ruby_task_helper/pull/4 to allow the task to correctly load the transport.
